components/fieldgroup/stories/fieldgroup.stories.js

```diff
@@ -7,6 +8,48 @@ import packageJson from "../package.json";
 import { FieldGroupSet } from "./fieldgroup.test.js";
 import { Template } from "./template.js";
 
+const inputType = (type) => type === "radio" ? [
```

Comment:
Oh nice. This is clever, and I wouldn't have thought to create a function like this here. I do have some other ideas for this, however. I'm happy to pair if this doesn't make sense!

I think we actually need to be editing `fieldgroup.template.js` instead of this stories file. If I'm understanding the card correctly, we want to be able to pass radio or checkbox to the `inputType` arg, and the story will handle the physical input changes. At one point, it was working (Josh linked to [the deploy preview](https://pr-2827--spectrum-css.netlify.app/preview/?path=/story/components-field-group--default&args=inputType:checkbox), and you can just switch the controls from radio to checkbox), and [line 44ish is I think what we want to try to do](https://github.com/adobe/spectrum-css/pull/2827/files#diff-9d5d7da10a73aa5f35dcd706223f6b2198005d038d0310251ee6d2fb9c673541): 

```
${inputType === "radio" ? 
  items.map((item) =>
	Radio({
	...item,
	isReadOnly,
	name: "field-group-example",
	customClasses: [`${rootClass}-item`],
  }, context)) 
  : items.map((item) =>
	Checkbox({
	...item,
	isReadOnly,
	customClasses: [`${rootClass}-item`],
  }, context))}
```

Or maybe we can edit the newer `renderContent` call? I had to import `Radio` and `Checkbox` into the template file instead.

```
${when(inputType, () => html`
	${renderContent(items, {
		callback: inputType == "radio" ? Radio : Checkbox,
		args: {
			label,
			isReadOnly,
			isRequired,
			customClasses,
	        }, context
	})}
	`,
)}
```

Reply:
![Kapture 2024-10-14 at 15 42 40](https://github.com/user-attachments/assets/b47a412b-4a5b-4b1a-84ed-e833bbd1aa2c)

Totally makes sense! I _did_ have to remove the `items` defaults in the story and set some in the template, but I'm open to another solution to that.

(Having them in the story rather than the template yielded an array of `null` objects.)